packages/headless/src/features/attached-results/attached-results-state.ts

```diff
@@ -0,0 +1,59 @@
+export interface AttachedResult {
+  /**
+   * Specifies the language of an attached Salesforce Knowledge Article.
+   */
+  articleLanguage?: string;
+  /**
+   * Specifies the version number of an attached Salesforce Knowledge Article.
+   */
+  articleVersionNumber?: string;
+  /**
+   * Specifies the publish status of an attached Salesforce Knowledge Article.
+   */
+  articlePublishStatus?: string;
+  /**
+   * The Id of the record (Case) the result is attached to.
+   */
+  caseId: string;
+  /**
+   * The Id of an attached Salesforce Knowledge Article.
+   */
+  knowledgeArticleId?: string;
+  /**
+   * The permanentId, the identifier of an attached result.
+   */
+  permanentId?: string;
```

Comment:
The casing of this seems to differ slightly (but annoyingly ;) ) from what the search API returns. Same comment for `urihash` below.

And it leaks in other places in this PR (for example when comparing `result.permanentid === attachedResult.permanentId` )

Similar to my above comment, I think in general this interface could benefit from `Pick`-ing properties from `Result`, and extending with the new properties that are specifically needed for attach result feature.

Reply:
Yea, the interface is different on purpose, the camelCase stuff is the data coming from the CRM AttachedResults, that's why this thing is so complicated... Because we need to convert the data from an external source to match the searchapi format.

I refactored it a little bit so it uses a lot more of the searchapi result format of data. It should be clearer I think?